Directly configuring a dest case fails at runtime

```scala
enum SourceLevel3 {
  case One(int: Int)
  case Two(str: String)
}

enum DestLevel3 {
  case One(int: Long)
   case Two(str: String)
}

val source = SourceLevel3.One(1)
val expected = DestLevel3.One(6)

source.into[DestLevel3].transform(
   Field.const(_.at[DestLevel3.One], DestLevel3.One(2L))
)
```

result:
```bash
sbt:root> ducktapeJVM/run
[info] running io.github.arainko.ducktape.test 
Exception in thread "sbt-bg-threads-7" java.lang.ExceptionInInitializerError
        at io.github.arainko.ducktape.test.main(Mode.scala)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at sbt.Run.invokeMain(Run.scala:144)
        at sbt.Run.execute$1(Run.scala:94)
        at sbt.Run.$anonfun$runWithLoader$5(Run.scala:121)
        at sbt.Run$.executeSuccess(Run.scala:187)
        at sbt.Run.runWithLoader(Run.scala:121)
        at sbt.Defaults$.$anonfun$bgRunTask$6(Defaults.scala:2038)
        at sbt.Defaults$.$anonfun$termWrapper$2(Defaults.scala:1977)
        at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
        at scala.util.Try$.apply(Try.scala:213)
        at sbt.internal.BackgroundThreadPool$BackgroundRunnable.run(DefaultBackgroundJobService.scala:367)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.RuntimeException: Unhandled case. This is most likely a bug in ducktape.
        at io.github.arainko.ducktape.test$.<clinit>(Mode.scala:191)
        ... 16 more
[success] Total time: 0 s, completed Oct 13, 2024, 12:59:01 PM
sbt:root> 
```

It works if we configure the innermost fields.

Yikes.

--- a/ducktape/src/main/scala/io/github/arainko/ducktape/internal/Plan.scala
+++ b/ducktape/src/main/scala/io/github/arainko/ducktape/internal/Plan.scala
@@ -172,40 +172,14 @@ private[ducktape] object Plan {
   }
 
   object Configured {
-    /*
-
-    */
     def from[F <: Fallible](plan: Plan[Erroneous, F], conf: Configuration[F], instruction: Configuration.Instruction[F])(using
       Quotes,
       Context
     ): Plan.Configured[F] =
-      (plan.source.tpe, plan.dest.tpe, conf.tpe) match {
-        case ('[src], '[dest], '[confTpe]) =>
-          
-          // val (source, dest) = conf match
-          //   case Configuration.Const(value, tpe) => 
-          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
-
-          //   case Configuration.CaseComputed(tpe, function) => 
-          //     Structure.Lazy.of[confTpe](plan.source.path) -> plan.dest
-
-          //   case Configuration.FieldComputed(tpe, function) =>
-          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
-
-          //   case Configuration.FieldReplacement(source, name, tpe) =>
-          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
-
-          //   case Configuration.FallibleConst(value, tpe) =>
-          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
-
-          //   case Configuration.FallibleFieldComputed(tpe, function) =>
-          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
-
-          //   case Configuration.FallibleCaseComputed(tpe, function) =>
-          //     Structure.Lazy.of[confTpe](plan.source.path) -> plan.dest
-          
-          // if side is Dest then we're operating either on a missing Source in which case
-          // val source = if instruction.side.isDest then Structure.Lazy.of[confTpe](plan.source.path) else plan.source
+      conf.tpe match {
+        case '[confTpe] =>
+          // if side is Source then we're operating on either a missing Source case or an override of that,
+          // which means the dest types need to be narrowed down to the exact type of this config
           val dest = if instruction.side.isSource then Structure.Lazy.of[confTpe](plan.dest.path) else plan.dest
           Plan.Configured(plan.source, dest, conf, instruction.span)
       }

--- a/ducktape/src/main/scala/io/github/arainko/ducktape/internal/Plan.scala
+++ b/ducktape/src/main/scala/io/github/arainko/ducktape/internal/Plan.scala
@@ -172,15 +172,42 @@ private[ducktape] object Plan {
   }
 
   object Configured {
+    /*
+
+    */
     def from[F <: Fallible](plan: Plan[Erroneous, F], conf: Configuration[F], instruction: Configuration.Instruction[F])(using
       Quotes,
       Context
     ): Plan.Configured[F] =
       (plan.source.tpe, plan.dest.tpe, conf.tpe) match {
         case ('[src], '[dest], '[confTpe]) =>
-          val source = if instruction.side.isDest then Structure.Lazy.of[confTpe](plan.source.path) else plan.source
+          
+          // val (source, dest) = conf match
+          //   case Configuration.Const(value, tpe) => 
+          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
+
+          //   case Configuration.CaseComputed(tpe, function) => 
+          //     Structure.Lazy.of[confTpe](plan.source.path) -> plan.dest
+
+          //   case Configuration.FieldComputed(tpe, function) =>
+          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
+
+          //   case Configuration.FieldReplacement(source, name, tpe) =>
+          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
+
+          //   case Configuration.FallibleConst(value, tpe) =>
+          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
+
+          //   case Configuration.FallibleFieldComputed(tpe, function) =>
+          //     plan.source -> Structure.Lazy.of[confTpe](plan.dest.path)
+
+          //   case Configuration.FallibleCaseComputed(tpe, function) =>
+          //     Structure.Lazy.of[confTpe](plan.source.path) -> plan.dest
+          
+          // if side is Dest then we're operating either on a missing Source in which case
+          // val source = if instruction.side.isDest then Structure.Lazy.of[confTpe](plan.source.path) else plan.source
           val dest = if instruction.side.isSource then Structure.Lazy.of[confTpe](plan.dest.path) else plan.dest
-          Plan.Configured(source, dest, conf, instruction.span)
+          Plan.Configured(plan.source, dest, conf, instruction.span)
       }
   }
 

--- a/ducktape/src/main/scala/io/github/arainko/ducktape/internal/PlanInterpreter.scala
+++ b/ducktape/src/main/scala/io/github/arainko/ducktape/internal/PlanInterpreter.scala
@@ -19,7 +19,7 @@ private[ducktape] object PlanInterpreter {
     plan match {
       case Plan.Upcast(_, _, _) => value
 
-      case Plan.Configured(_, _, config, _) =>
+      case Plan.Configured(_, dest, config, _) =>
         evaluateConfig(config, value)
 
       case Plan.BetweenProducts(sourceTpe, destTpe, fieldPlans) =>

--- a/ducktape/src/test/scala/io/github/arainko/ducktape/issues/Issue211Suite.scala
+++ b/ducktape/src/test/scala/io/github/arainko/ducktape/issues/Issue211Suite.scala
@@ -1,0 +1,24 @@
+package io.github.arainko.ducktape.issues
+
+import io.github.arainko.ducktape.*
+
+class Issue211Suite extends DucktapeSuite {
+  test("directly configuring a Dest case doesn't fail at runtime") {
+    enum SourceLevel3 {
+      case One(int: Int)
+      case Two(str: String)
+    }
+
+    enum DestLevel3 {
+      case One(int: Long)
+      case Two(str: String)
+    }
+
+    val source = SourceLevel3.One(1)
+    val expected = DestLevel3.One(6)
+
+    assertTransformsConfigured(source, expected)(
+      Field.const(_.at[DestLevel3.One], DestLevel3.One(6L))
+    )
+  }
+}

